hypex/dataset/datagenerator.py

```diff
@@ -0,0 +1,293 @@
+            m_with_dummy:
+                Array with features
+        """
+        means = np.random.uniform(-1, 1, self.num_features)
+        cov_mat = np.identity(self.num_features)
+        m = np.random.multivariate_normal(means, cov_mat, self.num_records)
+        m_with_dummy = to_categories(m, self.num_features, self.num_discrete_features)
+        return m_with_dummy
+
+    def _generate_info_cols(self):
+        """
+        Generate columns with information features for causal process
+
+        Returns:
+            info:
+                List with features
+        """
+        info = []
+        for i in range(self.num_info_cols):
+            if i % 2 == 0:
+                coefficient = 3 if i <= 2 else i
+                array = list(range(1, self.num_records * coefficient, coefficient))
+                random.shuffle(array)
+                info.append(array)
+            else:
+                unique_letters = list(set(string.ascii_letters.upper()))
+                letters = [random.choice(unique_letters) for _ in range(i * 2)]
+                info.append([letters[j] * ((i // 2) + 1) for j in np.random.choice(a=np.arange(i * 2),
+                                                                                   size=self.num_records)])
+        return info
+
+    def _generate_treatment_cols(self):
+        """
+        Generate columns with treatment values
+
+        Returns:
+            treatment:
+                Array with treatment
+        """
+        if self.is_treatment and self.num_targets == 1:
+            treatment = np.random.normal(0, self.sigma, (self.num_records, 1))
+        else:
+            treatment = np.random.normal(0, self.sigma, (self.num_records, self.num_targets))
+        treatment = np.digitize(treatment, bins=np.array([0, 0.5, 1])).astype(bool)
+        return treatment
+
+    def _generate_y_for_target(self, m_with_dummy):
+        """
+        Generate data for target field
+        Args:
+            m_with_dummy: Array with main features
+
+        Returns:
+            y0, y1: Arrays with values for target field
+        """
+        m_vector = m_with_dummy.dot(np.ones(self.num_features))
+        epsilon = np.random.multivariate_normal(mean=np.zeros(2),
+                                                cov=np.identity(2),
+                                                size=self.num_records)
+        y0 = m_vector + epsilon[:, 0]
+        y1 = self.sigma + m_with_dummy.dot(np.ones(self.num_features) * 2) + epsilon[:, 1]
+        return y0, y1
+
+    def _generate_outcome_and_full_data(self, m_with_dummy, treatment):
+        """
+        Generates outcome depending on treatment and create array with main numeric features
+        Args:
+            m_with_dummy: Array with main features
+            treatment: Array with treatment values
+
+        Returns:
+            data: Array with main numeric features
+        """
+        y0, y1 = self._generate_y_for_target(m_with_dummy)
```

Comment:
I still can't fully understand the logic, but it seems logical to generate y separately instead of hardcoding 2 in one function